src/hooks/useGridParticipants/useGridParticipants.ts

```diff
@@ -0,0 +1,94 @@
+import useVideoContext from '../useVideoContext/useVideoContext';
+
+// If an offscreen participant becomes the dominant speaker, we
+// move them onscreen to where the least recent onscreen dominant
+// speaker was located. We are able to order the speakers appropriately
+// by keeping track of the timestamp from when they became the newest
+// dominant speaker.
+
+interface OrderedParticipant {
+  participant: RemoteParticipant;
+  dominantSpeakerStartTime: number;
+}
+
+export default function useGridParticipants() {
+  const [orderedParticipants, setOrderedParticipants] = useState<OrderedParticipant[]>([]);
+  const { room } = useVideoContext();
+  const dominantSpeaker = useDominantSpeaker();
+  const { maxGridParticipants } = useAppState();
+
+  useEffect(() => {
+    if (dominantSpeaker !== null) {
+      setOrderedParticipants(prevParticipants => {
+        const newParticipantsArray = prevParticipants.slice();
+
+        const newDominantSpeaker = newParticipantsArray.find(p => p.participant === dominantSpeaker);
+        // it's possible that the dominantSpeaker is removed from the newParticipantsArray before they become null:
+        if (newDominantSpeaker) {
+          // update the participant's dominantSpeakerStartTime to when they became the new dominant speaker:
+          newDominantSpeaker.dominantSpeakerStartTime = Date.now();
+        } else {
+          return newParticipantsArray;
+        }
+
+        // Here we use maxGridParticipants - 1 since the localParticipant will always be the first in the grid
+        let maxOnScreenParticipants = maxGridParticipants - 1;
+        const onscreenParticipants = newParticipantsArray.slice(0, maxOnScreenParticipants);
+
+        // if the newest dominant speaker is not currently on screen, reorder the orderedParticipants array:
+        if (!onscreenParticipants.some(p => p.participant === dominantSpeaker)) {
+          // find the least recent dominant speaker by sorting the onscreen speakers by their dominantSpeakerStartTime:
+          const sortedOnscreenParticipants = onscreenParticipants.sort(
+            (a, b) => a.dominantSpeakerStartTime - b.dominantSpeakerStartTime
+          );
+          const leastRecentDominantSpeaker = sortedOnscreenParticipants[0];
+          const newDominantSpeakerWithStartTime = newParticipantsArray.find(p => p.participant === dominantSpeaker);
+
+          /** Reorder the onscreen participants */
+          // Temporarily remove the newest dominant speaker:
+          newParticipantsArray.splice(newParticipantsArray.indexOf(newDominantSpeakerWithStartTime!), 1);
+          // Remove the least recent dominant speaker and replace them with the newest:
+          newParticipantsArray.splice(
+            newParticipantsArray.indexOf(leastRecentDominantSpeaker),
+            1,
+            newDominantSpeakerWithStartTime!
+          );
+          // Add the least recent dominant speaker back into the  array at the end:
```

Comment:
I think this is the only discrepancy between iOS and web. Currently iOS moves the oldest dominant speaker on the first page to the start of the second page. This way the entire grid is ordered by dominant speaker time + connection time which seems intuitive. I am thinking this is how zoom works.

Reply:
@timmydoza I think what @timrozum did with iOS makes sense. However, it is kind of tricky with the current implementation for the pagination feature. Since we are filling each page with the number of `maxGridParticipants`, the first participant on the second page may actually be an `onscreenParticipant` from the first page. 

For instance, if a room has 13 participants, and `maxGridParticipants` is 9, the first participant on the second page will actually be the fifth participant from the first page (when on iOS the first participant on the second page would be the 10th participant). Did we come to a decision about updating  the `usePagination` hook to mirror was iOS does?